datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -79,17 +79,27 @@ func NewGraphSyncDataTransfer(parent context.Context, host host.Host, gs graphsy
 // if an incoming request does not match a previous push request, it returns an error.
 func (impl *graphsyncImpl) gsReqRecdHook(p peer.ID, request graphsync.RequestData) ([]graphsync.ExtensionData, error) {
 	var resp []graphsync.ExtensionData
-	chid, _, err := impl.getChannelIDAndData(request)
+
+	// if this is a push request the sender is us.
+	chid, err := impl.getChannelIDAndData(request, impl.peerID)
+	if err != nil {
+		return resp, err
+	}
 
 	extData := graphsync.ExtensionData{
 		Name: ExtensionDataTransfer,
 		Data: nil,
 	}
-	if err != nil {
-		return resp, err
-	}
-	if !impl.hasPushChannel(chid) {
-		return resp, errors.New("could not find push channel")
+	// if a push request, initiator & sender is us, ask for the channel with us as sender
+	if impl.getChannel(chid, impl.peerID) == datatransfer.EmptyChannelState {
+
+		// otherwise check if it's a pull request: the initiator is them
+		chid, err = impl.getChannelIDAndData(request, p)
+
+		// sender is still us
+		if impl.getChannel(chid, impl.peerID) == datatransfer.EmptyChannelState {
+			return resp, errors.New("could not find push channel")
```

Comment:
isn't this actually "count not find push channel or pull channel" (at this point -- cause you checked both)

Reply:
yes, I'll update the msg.